### PR TITLE
#gh11353: Allow bpartner filter by name, value or city name.

### PR DESCRIPTION
**Pull request by adi-stefan**
https://github.com/metasfresh/metasfresh/issues/11353

### CODE REVIEW

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/bpartner/filter/BPartnerSimpleFuzzySearchFilterProvider.java [hunk 1]
```diff
@@ -111,11 +123,14 @@ public String getSql(
...
 
-		// TODO: implement
-		// filter by: C_BPartner.Value, C_BPartner.Name, C_BPartner->C_BPartner_Location->C_Location.City etc (check the task description)
-		// use org.adempiere.db.DBConstants.FUNC_unaccent_string
+		return sqlOpts.getTableNameOrAlias() + "." + getSqlCriteria(sqlParamsOut, searchText);
 
-		throw new UnsupportedOperationException();
+	}
+
+	private String getSqlCriteria(final SqlParamsCollector sqlParamsOut, final String searchText)
```
Comment (teosarca) [inline backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/bpartner/filter/BPartnerSimpleFuzzySearchFilterProvider.java:130]: * consider using SqlAndParams
* don't forget using use org.adempiere.db.DBConstants.FUNC_unaccent_string when comparing strings. Basically the resulting SQL where clause shall look like 
```sql
unaccent_string(MyColumnName) ILIKE '%bla%bla%'
```

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/bpartner/filter/BPartnerSimpleFuzzySearchFilterProvider.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+@Component
+public class BPartnerSimpleFuzzySearchFilterProvider implements DocumentFilterDescriptorsProviderFactory, SqlDocumentFilterConverter
+{
+	private final ISysConfigBL sysConfigBL = Services.get(ISysConfigBL.class);
+
+	private static final String FILTER_ID = "BPartnerSimpleFuzzySearchFilterProvider";
+	private static final String PARAMETERNAME_SearchText = "searchText";
+	private static final String BPARTNER_SEARCH_SQL_TEMPLATE = ".C_Bpartner_ID IN (SELECT p.C_Bpartner_ID "
+			+ "FROM C_Bpartner p "
+			+ "LEFT OUTER JOIN C_Bpartner_Location pl ON p.C_Bpartner_ID = pl.C_Bpartner_ID AND pl.IsShipToDefault = 'Y' "
```
Comment (teosarca) [inline backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/bpartner/filter/BPartnerSimpleFuzzySearchFilterProvider.java:61]: IMHO we don't need `AND pl.IsShipToDefault = 'Y'`.

Instead I think we need pl.IsActive='Y'

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/bpartner/filter/BPartnerSimpleFuzzySearchFilterProvider.java (hunk 1 above)
Comment (adi-stefan) [reply backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/bpartner/filter/BPartnerSimpleFuzzySearchFilterProvider.java]: Changed to using SqlAndParams and unaccent_string(?,1)

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/bpartner/filter/BPartnerSimpleFuzzySearchFilterProvider.java (hunk 2 above)
Comment (adi-stefan) [reply backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/bpartner/filter/BPartnerSimpleFuzzySearchFilterProvider.java]: As discussed, `IsShipToDefault` is still required.

**Pull request merged**